sdk/go/common/util/fsutil/walkup.go

```diff
@@ -67,6 +67,44 @@ func WalkUp(path string, walkFn func(string) bool, visitParentFn func(string) bo
 	return "", nil
 }
 
+// WalkUpDirs visits each parent directory of path.
+//
+// For example, /a/b/c/foo.txt would visit in this order:
+//
+//	/a/b/c
+//	/a/b
+//	/a
+//	/
+//
+// If checkDir returns true the search terminates early and the matching dir is returned.
+func WalkUpDirs(path string, checkDir func(dir string) bool) (string, error) {
```

Comment:
Should we change `checkDir` to `func(dir string) (string, bool)` and make `WalkUpDirs` return that path, if the result is true?

Reply:
It could return a generic T result in that case, which does not have to be string. I'm on the fence here as the FP person in me wants this but I doubt this is idiomatic Go:

```
func WalkUpDirs[T any](path string, checkDir func(dir string) (T, bool)) (T, error)
```